lib/AST/SwiftNameTranslation.cpp

```diff
@@ -321,6 +307,51 @@ bool swift::cxx_translation::isVisibleToCxx(const ValueDecl *VD,
   return false;
 }
 
+bool swift::cxx_translation::isExposableToCxx(GenericSignature genericSig) {
+  // If there's no generic signature, it's fine.
+  if (!genericSig)
+    return true;
+
+  // FIXME: This should use getRequirements() and actually
+  // support arbitrary requirements. We don't really want
+  // to use getRequirementsWithInverses() here.
+  //
+  // For now, we use the inverse transform as a quick way to
+  // check for the "default" generic signature where each
+  // generic parameter is Copyable and Escapable, but not
+  // subject to any other requirements; that's exactly the
+  // generic signature that C++ interop supports today.
+  SmallVector<Requirement, 2> reqs;
+  SmallVector<InverseRequirement, 2> inverseReqs;
+  genericSig->getRequirementsWithInverses(reqs, inverseReqs);
+  if (!reqs.empty()) {
+    // Conformance requirements to marker protocols are okay.
+    for (const auto &req: reqs) {
+      if (req.getKind() != RequirementKind::Conformance)
+        return false;
+
+      auto proto = req.getProtocolDecl();
+      if (!proto->isMarkerProtocol())
+        return false;
+    }
+  }
+
+  // Allow Copyable and Escapable.
```

Comment:
You can simplify this further and remove the inverse transform. You just want to iterate over the requirements, ignoring marker protocols (which Copyable and Escapable are)